client/gutenberg/editor/components/header/opt-out-menu-item/index.jsx

```diff
@@ -0,0 +1,67 @@
+/** @format */
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * WordPress Dependencies
+ */
+import { MenuItem } from '@wordpress/components';
+import {
+	composeAnalytics,
+	recordGoogleEvent,
+	recordTracksEvent,
+	withAnalytics,
+	bumpStat,
+} from 'state/analytics/actions';
+import { setSelectedEditor } from 'state/selected-editor/actions';
+import getCurrentRoute from 'state/selectors/get-current-route';
+import { navigate } from 'state/ui/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+export const OptOutMenuItem = ( {
+	classicEditorRoute,
+	navigate: redirect,
+	optOut,
+	siteId,
+	translate,
+} ) => {
+	const switchToClassicEditor = () => {
+		optOut( siteId );
+		redirect( classicEditorRoute );
+	};
+	return (
+		<MenuItem onClick={ switchToClassicEditor }>
+			{ translate( 'Switch to Classic Editor' ) }
+		</MenuItem>
+	);
+};
+
+const optOut = siteId => {
+	return withAnalytics(
+		composeAnalytics(
+			recordGoogleEvent(
+				'Gutenberg Opt-Out',
+				'Clicked "Switch to the classic editor" in the editor More menu.',
+				'Opt-In',
+				false
+			),
+			recordTracksEvent( 'calypso_gutenberg_opt_in', {
+				opt_in: false,
+			} ),
+			bumpStat( 'gutenberg-opt-in', 'Calypso More Menu Opt Out' )
+		),
+		setSelectedEditor( siteId, 'classic' )
+	);
+};
+
+export default connect(
+	state => ( {
+		classicEditorRoute: `/${ getCurrentRoute( state ).replace( '/gutenberg/', '' ) }`,
```

Comment:
Folks aren't likely to hit this when the editor is loaded, but getCurrentRoute may return null.
```
export const getCurrentRoute = state => get( state, 'ui.route.path.current', null );
```